Src/Couchbase.Linq.IntegrationTests/BucketContextTests.cs

```diff
@@ -367,6 +367,26 @@ public void SubmitChanges_WhenDocsAreModifiedAndEndChangeTracking_DocumentIsNotM
             var doc = ClusterHelper.GetBucket("beer-sample").GetDocument<Beer>(((ITrackedDocumentNode)beer).Metadata.Id);
             Assert.AreNotEqual(beer.Abv, doc.Content.Abv);
         }
+
+        [Test]
+        public void Underlying_Bucket_can_be_used_to_execute_N1QL_query_directly()
+        {
+            // setup bucket context
+            var db = new BucketContext(ClusterHelper.GetBucket("beer-sample"));
+
+            // get a beer from the db
+            var beer = db.Query<Beer>().Select(b => N1QlFunctions.Meta(b)).First();
+
+            // execute N1QL to update the beer
+            var newBeerName = Guid.NewGuid().ToString();
+            var n1ql = string.Format("UPDATE `beer-sample` USE KEYS '{0}' SET name = '{1}';", beer.Id, newBeerName);
+            db.Bucket.Query<int>(n1ql);
```

Comment:
@jeffrymorris 

While the specific test being executed isn't really advantageous, I do like the support for this concept.  In particular, if the UPDATE statement was using a WHERE clause that updates multiple documents instead of USE KEYS.  In this case it will be more efficient to let the Query server handle reading and changing each document instead of on the client side.  On the client side, you would be forced to run a SELECT, then loop through the reuslts and run deletes.  This adds transferring all of the data from the Query node to the client unnecessarily.

Brant